### PR TITLE
Remove 2.361.x support

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Drop 2.361.x from plugin BOM

- Use 2.387.x in example
- Drop 2.361.x from plugin BOM

https://www.jenkins.io/security/advisory/2023-03-08/ notes a security vulnerability in Jenkins versions prior to 2.375.4.  Let's drop support for 2.361.x to reduce the testing load and to encourage users to upgrade and avoid security issues.

98% of the 96000 installations of git plugin 5.0 were already running 2.375.1 or newer as of 1 May 2023.  Over 40% of all git plugin installations were using 2.375.1 or newer as of 1 May 2023.  Retaining plugin BOM support for 2.361.x does not help our Jenkins users or developers enough to justify the effort maintaining it.

Users that are upgrading to within 6 months of the most recent release (about 1/3 of the total installed base) are upgrading both core and plugins.

### Testing done

Prep run locally and a plugin test was run locally

### Submitter checklist

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by jglick** [comment]
My only concern is that a lot of plugins use 2.361.x as their baseline, aligning with the switch to Java 11, so they are going to stop seeing BOM updates. Should we begin pushing them to 2.375.x?

**Comment by MarkEWaite** [comment]
> My only concern is that a lot of plugins use 2.361.x as their baseline, aligning with the switch to Java 11, so they are going to stop seeing BOM updates. Should we begin pushing them to 2.375.x?

I agree that many plugins use  2.361.x as their baseline.  I think that we should push them to 2.387.x unless they have a specific reason to remain at 2.375.x.

When this pull request is merged, I plan to start that change on the 18 plugins that I most actively maintain so that they will switch to 2.387.x as their minimum Jenkins version.

**Review by jglick — APPROVED** [review]

**Review by basil — APPROVED** [review]
The `README` has:

> The UC currently maintains releases for the [past 400 days](https://groups.google.com/g/jenkins-infra/c/LTrRUqkgeQA/m/UmQMD5gDAgAJ) so it is reasonable to retire BOMs for lines older than that, or otherwise when the number of accumulated version overrides becomes large.

In practice I don't think we've ever maintained a BOM line for more than 6-9 months due to the accumulation of small annoyances (such as having to add overrides to too many Dependabot PRs). That combined with the fact that we're apparently seeing (Pipeline? infrastructure?) scalability issues testing more than 4 lines might lead us to simply admit that the 400 day ideal in the README is too impractical:

> The UC currently maintains releases for the [past 400 days](https://groups.google.com/g/jenkins-infra/c/LTrRUqkgeQA/m/UmQMD5gDAgAJ), so any BOM lines older than that should certainly be retired. Practical concerns usually lead to the retiring of a given line after 6-9 months.

**Comment by jglick** [comment]
Such an edit would be fine with me.

File: README.md [hunk 1]
```diff
@@ -186,9 +186,10 @@ The CI build (or just `mvn test -P2.nnn.x`) will fail if some managed plugins ar
 [This script](https://gist.github.com/jglick/0a85759ea65f60e107ac5a85a5032cae)
 is a handy way to find the most recently released plugin version compatible with a given line,
 according to the `jenkins-infra/update-center2`.
-The UC currently maintains releases for the [past 400 days](https://groups.google.com/g/jenkins-infra/c/LTrRUqkgeQA/m/UmQMD5gDAgAJ)
-so it is reasonable to retire BOMs for lines older than that,
-or otherwise when the number of accumulated version overrides becomes large.
+
+The [developer documentation](https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/) recommends the last releases of each of the previous two LTS baselines.
+BOMs for the current LTS release and two prior LTS releases are typically retained.
+BOMs older than the two prior LTS releases will generally be retired in better manage evaluation costs and maintenance efforts.
```
Comment (jglick) [inline README.md:192]:
```suggestion
BOMs older than the two prior LTS releases will generally be retired in order to better manage evaluation costs and maintenance efforts.
```

**Comment by jglick** [comment]
BTW I think you can remove `full-test` now?

**Comment by MarkEWaite** [comment]
> BTW I think you can remove `full-test` now?

The first run of the pull request only evaluated the portion that is evaluated for dependabot pull requests.  I assumed that I needed `full-test` to have it run the same steps that will be run on the master branch when it is re-run by a maintainer.  Have I misunderstood the use of `full-test`?

**Comment by jglick** [comment]
> I needed `full-test` to have it run the same steps that will be run on the master branch when it is re-run by a maintainer.

Right, I just mean once you have successfully tested the substantive changes using `full-test`, you can turn it off while doing stuff like editing `README.md`.

**Comment by timja** [comment]
I get the same error @MarkEWaite

**Comment by jglick** [comment]
@MarkEWaite it appears that #1896 introduced the regression. I am not sure why, and I am not sure why CI builds succeeded prior to now. I suspect that there is something nondeterministic about its build (using some resource from the Internet which changed recently).

**Comment by MarkEWaite** [comment]
Thanks!  Would you propose the revert of #1896?

**Comment by jglick** [comment]
I have no idea. Perhaps https://github.com/jenkinsci/bootstrap5-api-plugin/commit/df0d4e4e5daecd996ef5631db74c052b14099567 is the ultimate cause? @uhafner any comment?

**Comment by MarkEWaite** [comment]
> I have no idea. Perhaps [jenkinsci/bootstrap5-api-plugin@df0d4e4](https://github.com/jenkinsci/bootstrap5-api-plugin/commit/df0d4e4e5daecd996ef5631db74c052b14099567) is the ultimate cause? @uhafner any comment?

Fascinating that a commit from 26 March 2023 ( #1896) shows a failure.  

I see the same failure in on my development machine.  With the commit prior to #1896 , the `PLUGINS=bootstrap5-api bash local-test.sh` command works on my machine.  With that commit, it fails.  I'm running nodeJS version v14.16.1 but don't understand how we missed those failures in the intervening bill of material releases.

**Comment by uhafner** [comment]
Is the problem with a specific commit of the bootstrap plugin or with the latest one? bootstrap5-api@5.2.2 is some weeks old.

**Comment by MarkEWaite** [comment]
> Is the problem with a specific commit of the bootstrap plugin or with the latest one? bootstrap5-api@5.2.2 is some weeks old.

The original failure (and most interesting) is that the plugin bill of materials fails to complete its release process because the current bootstrap5-api plugin version in the plugin BOM master branch is failing when it runs tests.  The same tests can be run from a clone of this repository with the command:

> PLUGINS=bootstrap5-api bash local-test.sh

I don't understand why those tests are failing.  If you can run the same command from a fork of this repository, you may recognize something in the failure that is not apparent to me.

**Comment by MarkEWaite** [comment]
> Is the problem with a specific commit of the bootstrap plugin or with the latest one? bootstrap5-api@5.2.2 is some weeks old.

The more surprising problem (but lower priority) is that the command 

> PLUGINS=bootstrap5-api bash local-test.sh

fails on the commits since #1896 (late March 2023) , yet we've released multiple times since #1896 without detecting that failure.  I like the suggestion from @jglick that there must be something in the downloaded components that has changed in order to cause tests to fail that were previously passing.

**Comment by uhafner** [comment]
The bootstrap plugin uses `npm` as a side process do download the actual bootstrap files from the `npm` registry.  

I also noticed some compile errors while I switched to the lastet `npm` version. These went away after cleaning the `npm _modules` folder on disk. Maybe it helps to always clean those folders before building with maven?

> PLUGINS=bootstrap5-api bash local-test.sh

I'll try to run that locally as well.

**Comment by kerogers-cloudbees** [comment]
Based on some quick searching, it may be necessary to import `variables-dark` before importing `maps` at https://github.com/jenkinsci/bootstrap5-api-plugin/blob/75d2a8cd7f5baf8eed98da5f37341cc084691417/etc/bootstrap-custom-build.scss#L22

**Comment by jtnord** [comment]
What I noticed in the [repo](https://github.com/jenkinsci/bootstrap5-api-plugin/tree/75d2a8cd7f5baf8eed98da5f37341cc084691417) was that there is no `.lock` file so versions would IIUC float to "compatable versions by default - hence causing the non-determinism?

**Pull request merged**